ui/raidboss/emulator/EmulatorCommon.ts

```diff
@@ -0,0 +1,127 @@
+import { LocaleNetRegex } from '../../../resources/translations';
+import NetRegexes from '../../../resources/netregexes';
+
+// Disable ban-types for cloneData as it needs to work on raw objects for performance reasons.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type DataType = { [key: string]: any } | null;
```

Comment:
I can't come up with any better way to do this, and I think it's probably cleaner to use `any` here than to use `Data` from `data.d.ts` with a `[key: string]: any` field on it.  So, this all seems good to me, thanks!

Reply:
Yeah, I struggled with this one for a while, originally I wanted to just use `object` and disable the warning for that because that's what it's actually working on, but then it complained on every line that tried to access the object's properties with `data[<some_variable>]` so ultimately I went with this.